src/freenas/etc/ix.rc.d/ix-netif

```diff
@@ -219,6 +219,20 @@ _interface_config()
 
 netif_start()
 {
+	# before we make any changes to the interfaces, we disable
+	# carp advertisements. This is reenabled later in boot (ix-haready)
+	/sbin/sysctl net.inet.carp.allow=0 > /dev/null 2>&1
+
+	# enable CARP preemption
+	/sbin/sysctl net.inet.carp.preempt=1 > /dev/null 2>&1
+
+	# setting ifdown and senderr sysctl values to 0 prevents CARP
+	# from automatically changing the advskew and therefore the
+	# priority of the interface. TrueNAS's HA code handles this
+	# logic.
+	/sbin/sysctl net.inet.carp.ifdown_demotion_factor=0 > /dev/null 2>&1
+	/sbin/sysctl net.inet.carp.senderr_demotion_factor=0 > /dev/null 2>&1
```

Comment:
So the problem I have with this, is that ix-netif is the last place I would look for default sysctl values.

I would be happier if we kept sysctl.conf

Reply:
fair enough